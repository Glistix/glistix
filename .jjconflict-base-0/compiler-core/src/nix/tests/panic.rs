use crate::assert_nix;

#[test]
fn bare() {
    assert_nix!(
        r#"
fn go() {
  panic
}
"#,
    );
}

#[test]
fn panic_as() {
    assert_nix!(
        r#"
fn go() {
  let x = "wibble"
  panic as x
}
"#,
    );
}

#[test]
fn as_expression() {
    assert_nix!(
        r#"
fn go(f) {
  let boop = panic
  f(panic)
}
"#,
    );
}

#[test]
fn pipe() {
    assert_nix!(
        r#"
fn go(f) {
  f |> panic
}
"#,
    );
}

#[test]
fn sequence() {
    assert_nix!(
        r#"
fn go(at_the_disco) {
  panic
  at_the_disco
}
"#,
    );
}

#[test]
fn case() {
    assert_nix!(
        r#"
fn go(x) {
  case x {
    _ -> panic
  }
}
"#,
    );
}

#[test]
fn panic_with_call() {
    assert_nix! {
        r#"
fn go(x) {
  panic as x(5, "abc")
}
"#
    }
}
