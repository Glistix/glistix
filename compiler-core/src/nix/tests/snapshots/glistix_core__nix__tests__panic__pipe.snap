---
source: compiler-core/src/nix/tests/panic.rs
expression: "\nfn go(f) {\n  f |> panic\n}\n"
---
let
  inherit (builtins.import ./../gleam.nix) makeError;
  
  go =
    f:
    let
      _pipe = f;
    in
    (builtins.throw
        (makeError "panic" "my/mod" 3 "go" "panic expression evaluated" { }))
      _pipe;
in
{ }
